package.json

```diff
@@ -77,8 +77,8 @@
     "@nextcloud/logger": "^2.2.1",
     "@nextcloud/router": "^3.0.0",
     "@nextcloud/vue-select": "^3.25.0",
-    "@vueuse/components": "^10.0.2",
-    "@vueuse/core": "^10.1.2",
+    "@vueuse/components": "^10.7.2",
```

Comment:
Doesn't it include the bug, because the fix was not released?

Reply:
yes it needs to be:
`"@vueuse/components": "~10.7.2 || ^10.8.1"`

Reply:
`npm ci` doesn't use lock from the dependencies, only the root.

Reply:
I'm not sure about this!
That's why doing `npm i`  or deleting the lockfile on server generates so many changes.
I just tested it, and installing this PR does in fact instal the proper `10.7.x` version on server too :shrug:

Reply:
I am sure about it.

Lock files of the dependencies are not used, only on the root.

It installs `10.7.x` on server after `npm ci` if the server has `10.7.x` in the lock file.

But having `10.7.x` in the lock file of `@nextcloud/vue` does nothing when using the package.
Updating dependencies on server will install `10.8.x`.

Otherwise we would always have tons of packages
```
root
- A
  - package -- lib:^1.0.0
  - package-lock -- lib:1.2.3
- B
  - package -- lib:^1.0.0
  - package-lock -- lib:1.2.4
- C
  - package -- lib:^1.0.0
  - package-lock -- lib:1.3.0
```

The root above is supposed to have only one version of `lib`, any above `1.0.0`, not 3 versions based on 3 lock files.

> I just tested it

Didn't you test with `npm link`? Asking, because `npm link` doesn't simulate dependency resolving. It uses what you have installed in your local `@nextcloud/vue`.

Reply:
The lockfile is only for development and testing, it is not even shipped with packages.
Meaning NPM will only consider `package.json` for when calculating the dependency tree.

Thus we need to specify the versions we support in  the `package.json` so npm can decide to duplicate a dependency or not.

Reply:
Moreover this would not even be possible, imagine your dependency is using yarn or something other, they use different lock file formats. Only the `package.json` is standardized and considered for dependency lookup.

Reply:
We can also drop that PR, I fixed server manually :)

Reply:
But fixing server doesn't fix apps. We can change `^10.7.2` to `~10.7.2` to avoid `10.8` and update it when the fix will be released, or use the string from `@susnux`, both will guarantee that everything works for the library users.

Reply:
I would prefer to include the `1.8.1+` versions because otherwise users are bound to outdated library versions while there is no good reason to do so

Reply:
Feel free to edit this PR :)